src/transport/manager.c

```diff
@@ -62,20 +62,24 @@ z_result_t _z_new_transport_client(_z_transport_t *zt, _z_string_t *locator, _z_
     return ret;
 }
 
-z_result_t _z_new_transport_peer(_z_transport_t *zt, _z_string_t *locator, _z_id_t *local_zid) {
+static z_result_t _z_new_transport_peer(_z_transport_t *zt, _z_string_t *locator, _z_id_t *local_zid, bool is_open) {
     z_result_t ret = _Z_RES_OK;
     // Init link
     _z_link_t zl;
     memset(&zl, 0, sizeof(_z_link_t));
     // Listen link
-    ret = _z_listen_link(&zl, locator);
+    if (is_open) {
```

Comment:
IMHO parameter `is_open` passing through parameters is not very informative, as is its initialization in _z_open, It will more clear if it will be enum `_z_open_mode` with two values: `_Z_OPEN_MODE_CONNECT`/`_Z_OPEN_MODE_LISTEN`.